File: go/internal/agent/setup/generate.go
package setup

import (
	"context"
	"strings"

	"google.golang.org/adk/model"
	"google.golang.org/genai"
)

// GenerateText runs a single non-streaming completion: system is the instruction,
// user is the prompt, and the concatenated text response is returned. It lets
// callers outside this package use an LLM without importing genai directly.
func GenerateText(ctx context.Context, llm model.LLM, system, user string) (string, error) {
	req := &model.LLMRequest{
		Contents: []*genai.Content{UserText(user)},
		Config:   &genai.GenerateContentConfig{SystemInstruction: UserText(system)},
	}
	var sb strings.Builder
	for resp, err := range llm.GenerateContent(ctx, req, false) {
		if err != nil {
			return "", err
		}
		if resp.Content != nil {
			sb.WriteString(contentText(resp.Content))
		}
	}
	return sb.String(), nil
}

File: kotlin/src/main/kotlin/com/automation/agent/agent/setup/Generate.kt
package com.automation.agent.agent.setup

import com.google.adk.kt.models.LlmRequest
import com.google.adk.kt.models.Model
import com.google.adk.kt.types.GenerateContentConfig
import kotlinx.coroutines.flow.fold

/**
 * Runs a single non-streaming completion: [system] is the instruction, [user] is the prompt, and
 * the concatenated text response is returned. It lets callers outside this package use a [Model]
 * without touching the genai types directly.
 *
 * A failing model throws from the underlying [Flow]; the exception propagates out of this suspend
 * function.
 */
suspend fun generateText(llm: Model, system: String, user: String): String {
    val req =
        LlmRequest(
            contents = listOf(userText(user)),
            config = GenerateContentConfig(systemInstruction = userText(system)),
        )
    return llm.generateContent(req, stream = false).fold(StringBuilder()) { sb, resp ->
        sb.append(contentText(resp.content))
    }.toString()
}
